File: api/Core/Model/ProductGroup.cs
using System;

namespace Xcianify.Core.Model
{
    public class ProductGroup : BaseModel
    {
        public string Code { get; set; } = string.Empty;
        public string Level { get; set; } = string.Empty;
        public string ProductGroupName { get; set; } = string.Empty;
        public string Unit { get; set; } = "NOS";
        public int? SalesDivisionCode { get; set; }
        public string UomForMls { get; set; } = "NOS";
        public decimal ConversionFactor { get; set; } = 1.00000000m;
        public string ConversionFactorUnit { get; set; } = "NOS/NOS";
        public string? CostCentreCode { get; set; }
        public bool IsClosed { get; set; } = false;
        public string? UpdatedBy { get; set; }
        public DateTime? UpdatedTimestamp { get; set; }
        public string RevNo { get; set; } = "00938";
        public bool IsActive { get; set; } = false;
    }
} 

File: pharma-erp-3/types/product-group.ts
export interface ProductGroup {
  id: number;
  code: string;
  level: string;
  productGroupName: string;
  unit: string;
  salesDivisionCode?: number;
  uomForMls: string;
  conversionFactor: number;
  conversionFactorUnit: string;
  costCentreCode?: string;
  isClosed: boolean;
  updatedBy?: string;
  updatedTimestamp?: string;
  revNo: string;
  isActive: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateProductGroupData {
  code: string;
  level: string;
  productGroupName: string;
  unit: string;
  salesDivisionCode?: number;
  uomForMls: string;
  conversionFactor: number;
  conversionFactorUnit: string;
  costCentreCode?: string;
  isClosed: boolean;
  revNo: string;
  isActive: boolean;
}

export interface UpdateProductGroupData {
  code: string;
  level: string;
  productGroupName: string;
  unit: string;
  salesDivisionCode?: number;
  uomForMls: string;
  conversionFactor: number;
  conversionFactorUnit: string;
  costCentreCode?: string;
  isClosed: boolean;
  revNo: string;
  isActive: boolean;
}

export interface ProductGroupFilter {
  search?: string;
  level?: string;
  isActive?: boolean;
  page?: number;
  pageSize?: number;
} 